clcache.py

```diff
@@ -164,33 +164,43 @@ def clean(self, stats, maximumSize):
         except ImportError:
             walker = os.walk
 
-        objects = [os.path.join(root, "object")
-                   for root, folder, files in walker(self.objectsDir)
-                   if "object" in files]
+        objectInfos = []
+        for x in xrange(16):
+            try:
+                objects = [os.path.join(root, "object")
+                       for root, folder, files in walker(self.objectsDir)
+                       if "object" in files]
+
+                objectInfos = [(os.stat(fn), fn) for fn in objects]
+                break
+            except WindowsError:
+                print( "exception during \"clean\", retrying" )
+
+        if not objectInfos:
+            return
 
-        objectInfos = [(os.stat(fn), fn) for fn in objects]
         objectInfos.sort(key=lambda t: t[0].st_atime)
 
         # compute real current size to fix up the stored cacheSize
         currentSize = sum(x[0].st_size for x in objectInfos)
 
         removedItems = 0
         for stat, fn in objectInfos:
-            rmtree(os.path.split(fn)[0])
+            rmtree(os.path.split(fn)[0], ignore_errors=True)
```

Comment:
again: `rmtree` may throw when the directory has been deleted from elsewhere. better gracefully ignore the error than fail (especially on buildservers)